q2cli/handlers.py

```diff
@@ -504,6 +504,11 @@ def get_type(self):
                 return mapping['Float']
             else:
                 raise NotImplementedError()
+        # Specify 'Set'/'List' because 'Dict' will require more modifications
+        elif self.ast['name'] in ['Set', 'List']:
+            # n-`primitive`s can not be unioned, so ignore trailing values
+            field, *_ = self.ast['fields']
+            return mapping[field['name']]
```

Comment:
This should handle the same predicate logic as above. So ideally we would return a `click.Choice` or `click.IntRange` if appropriate.

Reply:
I would create a helper `_convert_primitive` which does the mess of logic that `get_type` usually does, then you can make `get_type` just check:

```python
if self.ast['name'] in 'Set', 'List':
    field, = self.ast['fields']
    return _convert_primitive(field)
else:
    return _convert_primitive(self.ast)
```